pkg/agent/hostagent.go

```diff
@@ -273,6 +272,42 @@ func (c *HostAgent) updateYAML(configType, yamlPath string) error {
 		}
 	}
 
+	// Adding docker_stats settings in the config, if docker endpoint is valid
+	dockerSocketPath := strings.Split(c.DockerEndpoint, "//")
+
+	if len(dockerSocketPath) == 2 && isSocketFn(dockerSocketPath[1]) {
+
+		if rec, o := apiYAMLConfig["receivers"].(map[string]interface{}); o {
+			rec["docker_stats"] = map[string]interface{}{
+				"collection_interval": "5s",
+				"endpoint":            c.DockerEndpoint,
+				"timeout":             "20s",
+				"api_version":         "1.24",
+			}
+			apiYAMLConfig["receivers"] = rec
+
+		} else {
+			c.logger.Error("failed to parse receivers config")
+		}
+
+		if service, o := apiYAMLConfig["service"].(map[string]interface{}); o {
```

Comment:
Let's avoid using typecasting. Create appropriate structures that you can unmarshal into.